sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -1574,6 +1693,40 @@ private async Task RunProcessingAsync(CancellationToken cancellationToken)
                                                   string operationDescription,
                                                   CancellationToken cancellationToken) => Task.Run(() => OnProcessingErrorAsync(exception, partition, operationDescription, cancellationToken), CancellationToken.None);
 
+        /// <summary>
+        ///   Performs the actions needed to validate the connection to the requested
+        ///   Event Hub.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the validation.</param>
+        ///
+        private async Task ValidateEventHubsConnectionAsync(CancellationToken cancellationToken = default)
+        {
+            // Validate that the Event Hubs connection is valid by querying properties of the Event Hub.
+            // This is core functionality for the processor to discover partitions and validates read access.
+
+            var connection = CreateConnection();
+            await using var connectionAwaiter = connection.ConfigureAwait(false);
+            await connection.GetPropertiesAsync(RetryPolicy, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        ///   Performs the actions needed to validate the connection to the storage
+        ///   provider for checkpoints and ownership.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the validation.</param>
+        ///
+        private async Task ValidateStorageConnectionAsync(CancellationToken cancellationToken)
+        {
+            // Because the processor does not have knowledge of what storage implementation is in use,
+            // it cannot perform any specific in-depth validations.  Use the standard checkpoint query
+            // for an invalid partition; this should ensure that the basic storage connection can be made
+            // and that a read operation is valid.
+
+            await GetCheckpointAsync("invalid", cancellationToken).ConfigureAwait(false);
```

Comment:
Is the intent that this throws when the storage connection is invalid?

Reply:
Yes.  This is intended to validate that the connection string is correct, the container exists, and there are rights to query the container.   We're deliberately using an invalid partition identifier because we don't want to actually download a real checkpoint; the "Not Found" is good enough to prove connectivity.